internal/api/mail.go

```diff
@@ -464,7 +529,32 @@ func (a *API) sendEmailChange(r *http.Request, tx *storage.Connection, u *models
 
 	u.EmailChangeConfirmStatus = zeroConfirmation
 	now := time.Now()
-	if err := mailer.EmailChangeMail(r, u, otpNew, otpCurrent, referrerURL, externalURL); err != nil {
+
+	if config.Hook.CustomEmailProvider.Enabled {
+		// TODO: add in OTP New and additional fields
+		input := hooks.CustomEmailProviderInput{
+			UserID:          u.ID,
+			Email:           u.Email.String(),
+			OTP:             otpCurrent,
+			EmailActionType: mail.EmailChangeCurrentVerification,
+		}
+		output := hooks.CustomEmailProviderOutput{}
+		err = a.invokeHTTPHook(ctx, r, &input, &output, config.Hook.CustomEmailProvider.URI)
+		currentEmail := u.GetEmail()
+		if config.Mailer.SecureEmailChangeEnabled && currentEmail != "" {
```

Comment:
We can potentially merge the two invocations into one by adding additional parameters though that makes handling on the function slightly more cumbersome as there's a need to check for two different paramers `otp` and `newOPT` or similar.